### ISSUE TITLE
com.google.analytics.enhanced-ecommerce required fields

### ISSUE THREAD

**Issue by elliotlings** [snowplow/iglu-central#921]
Do you know why the fields in this schema doesn't allow null https://github.com/snowplow/iglu-central/blob/master/schemas/com.google.analytics.enhanced-ecommerce/productFieldObject/jsonschema/1-0-0 and this schema does? https://github.com/snowplow/iglu-central/blob/master/schemas/com.google.analytics.measurement-protocol/product/jsonschema/1-0-0

When calling 
`
window.snowplow('addEnhancedEcommerceProductContext',
  'P12345', // The ID
  'Android Warhol T-Shirt', // The name
  'Search Results', // The list
  'Google', // The brand
  'Apparel/T-Shirts', // The category
  'Black', // The variant
  1 // The quantity
);
`

you may pass null for a parameter and therefore won't pass validation.

**Comment by BenFradet** [comment]
hey, I think this is a blunder on the enhanced-ecommerce's side, wdyt @dilyand ?

**Comment by dilyand** [comment]
Yes, I agree. All the product fields are optional in the Measurement Protocol's [parameter reference docs](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#enhanced-ecomm), which makes https://github.com/snowplow/iglu-central/blob/master/schemas/com.google.analytics.enhanced-ecommerce/productFieldObject/jsonschema/1-0-0 too restrictive.

**Comment by BenFradet** [comment]
Cool thanks, @elliotlings feel free to open a PR for a more relaxed schema :+1:

**Comment by elliotlings** [comment]
I looked into this further and it may not be too much of a problem as I first thought. It turns out if you pass `undefined` instead of `null` they are stripped out by `JSON.stringify` and therefore won't be sent up. And as the schema doesn't specify any required properties, missing values won't cause an issue.
